test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/NetCorePackageReferenceProjectTests.cs

```diff
@@ -447,6 +456,1909 @@ public async Task GetInstalledVersion_WithAssetsFile_ChangingPackageSpec_Returns
+                    var projectFullPaths = new string[numberOfProjects];
+                    var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                    var nuGetPackageManager = new NuGetPackageManager(
+                        sourceRepositoryProvider,
+                        NullSettings.Instance,
+                        testSolutionManager,
+                        deleteOnRestartManager);
+
+                    var testNuGetProjectContext = new TestNuGetProjectContext();
+                    var netCorePackageReferenceProjects = new List<NetCorePackageReferenceProject>();
+                    var prevProj = string.Empty;
+                    PackageSpec prevPackageSpec = null;
+
+                    // Create projects
+                    for (var i = numberOfProjects-1; i >= 0; i--)
+                    {
+                        var projectName = $"project{i}";
+                        var directory = TestDirectory.Create();
+                        projectDirectories.Add(directory);
+                        var projectFullPath = Path.Combine(directory.Path, projectName + ".csproj");
+                        var project = CreateTestNetCorePackageReferenceProject(projectName, projectFullPath, projectCache);
+
+                        // We need to treat NU1605 warning as error.
+                        project.IsNu1605Error = true;
+                        netCorePackageReferenceProjects.Add(project);
+                        testSolutionManager.NuGetProjects.Add(project);
+
+                        //Let new project pickup my custom package source.
+                        project.ProjectLocalSources.AddRange(sources);
+                        var projectNames = GetTestProjectNames(projectFullPath, projectName);
+                        var packageSpec =  GetPackageSpec(
+                            projectName,
+                            projectFullPath,
+                            packageA_Version100.Version);
+
+                        if (prevPackageSpec != null)
+                        {
+                            packageSpec = packageSpec.WithTestProjectReference(prevPackageSpec);
+                        }
+
+                        // Restore info
+                        var projectRestoreInfo = ProjectJsonTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                        projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                        projectCache.AddProject(projectNames, projectAdapter, project).Should().BeTrue();
+                        prevProj = projectFullPath;
+                        packageSpecs[i] = packageSpec;
+                        prevPackageSpec = packageSpec;
+                        projectFullPaths[i] = projectFullPath;
+                    }
+
+                    var initialInstalledPackages = await netCorePackageReferenceProjects[numberOfProjects-1].GetInstalledPackagesAsync(CancellationToken.None);
```

Comment:
nit: this is not used here.

Reply:
Added assert for this. Make sure initial installed packages number is 1.